### PR TITLE
GO-2108: fix json import/export issues

### PR DESCRIPTION
**Pull request by AnastasiaShemyakinskaya**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### Description

Currently there is a problem with importing single page to Anytype, because we don't save relations, options and object type of exported object. So eventually we import object without its relations, options and object type (if it is custom). 

I added saving of relations, options and object types during export

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Related Tickets & Documents
https://linear.app/anytype/issue/GO-2108/fix-json-importexport-issues

### Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by fat-fellow — APPROVED** [review]

File: core/block/export/export.go [hunk 1]
```diff
@@ -523,3 +530,152 @@ func (e *export) cleanupFile(wr writer) {
...
+			},
+			{
+				RelationKey: bundle.RelationKeyIsDeleted.String(),
+				Condition:   model.BlockContentDataviewFilter_Equal,
+				Value:       pbtypes.Bool(false),
+			},
+		},
+	})
+	if err != nil {
+		log.Errorf("failed to query relation options: %s", err)
```
[the remark this answers is not in the document]
Comment (AnastasiaShemyakinskaya) [reply core/block/export/export.go]: Fixed

File: core/block/export/export.go [hunk 2]
```diff
@@ -523,3 +530,152 @@ func (e *export) cleanupFile(wr writer) {
...
+	id := value.GetStringValue()
+	filter = &model.BlockContentDataviewFilter{
+		RelationKey: bundle.RelationKeyId.String(),
+		Condition:   model.BlockContentDataviewFilter_Equal,
+		Value:       pbtypes.String(id),
+	}
+	return filter
+}
+
+func (e *export) getObjectType(details *types.Struct) []database.Record {
```
[the remark this answers is not in the document]
Comment (AnastasiaShemyakinskaya) [reply core/block/export/export.go]: Fixed

File: core/block/export/export.go [hunk 3]
```diff
@@ -523,3 +530,152 @@ func (e *export) cleanupFile(wr writer) {
...
+	}
+	return &relation[0]
+}
+
+func (e *export) addRelationAndOptions(relation *database.Record, value *types.Value, derivedObjects []database.Record) []database.Record {
+	derivedObjects = e.addRelation(relation, derivedObjects)
+	format := relation.Get(bundle.RelationKeyRelationFormat.String()).GetNumberValue()
+	if format == float64(model.RelationFormat_tag) || format == float64(model.RelationFormat_status) {
+		relationOptions := e.getRelationOptions(value)
+		derivedObjects = lo.Union(derivedObjects, relationOptions)
```
[the remark this answers is not in the document]
Comment (AnastasiaShemyakinskaya) [reply core/block/export/export.go]: Fixed

File: core/block/export/export.go [hunk 4]
```diff
@@ -523,3 +530,152 @@ func (e *export) cleanupFile(wr writer) {
...
+	}
+	if len(relation) == 0 {
+		return nil
+	}
+	return &relation[0]
+}
+
+func (e *export) addRelationAndOptions(relation *database.Record, value *types.Value, derivedObjects []database.Record) []database.Record {
+	derivedObjects = e.addRelation(relation, derivedObjects)
+	format := relation.Get(bundle.RelationKeyRelationFormat.String()).GetNumberValue()
```
[the remark this answers is not in the document]
Comment (AnastasiaShemyakinskaya) [reply core/block/export/export.go]: Fixed

File: core/block/export/export.go [hunk 5]
```diff
@@ -523,3 +530,152 @@ func (e *export) cleanupFile(wr writer) {
...
+	objectTypeDetails := e.getObjectType(details)
+	derivedObjects = lo.Union(derivedObjects, objectTypeDetails)
+
+	return derivedObjects
+}
+
+func (e *export) getRelation(key string) *database.Record {
+	uniqueKey, err := domain.NewUniqueKey(smartblock.SmartBlockTypeRelation, key)
+	if err != nil {
+		log.Errorf("failed to get relation unique key %s: %s", key, err)
```
[the remark this answers is not in the document]
Comment (AnastasiaShemyakinskaya) [reply core/block/export/export.go]: Fixed

File: core/block/export/export.go [hunk 6]
```diff
@@ -523,3 +530,152 @@ func (e *export) cleanupFile(wr writer) {
...
+func (e *export) processObject(object database.Record, derivedObjects []database.Record) []database.Record {
+	details := object.Details
+	for key, value := range details.Fields {
+		relation := e.getRelation(key)
+		if relation != nil {
+			derivedObjects = e.addRelationAndOptions(relation, value, derivedObjects)
+		}
+	}
+
+	objectTypeDetails := e.getObjectType(details)
```
[the remark this answers is not in the document]
Comment (AnastasiaShemyakinskaya) [reply core/block/export/export.go]: Fixed

File: core/block/export/export.go [hunk 7]
```diff
@@ -528,3 +539,154 @@ func (e *export) cleanupFile(wr writer) {
...
+		err            error
+	)
+
+	for _, object := range objects {
+		derivedObjects, err = e.processObject(object, derivedObjects)
+		if err != nil {
+			return nil, err
+		}
+	}
+	derivedObjects = lo.Union(derivedObjects)
```
Comment (deff7) [inline core/block/export/export.go:555]: redundant

Comment (AnastasiaShemyakinskaya) [reply core/block/export/export.go]: Fixed

**Pull request merged**